docs/Quick-start-and-introduction.md

```diff
@@ -145,12 +145,12 @@ if ($result->isValid()) {
 
 The result sets objects code returned from `getCode()` will match one of these codes. The codes are implemented as class constants in the Result object.
 
- * **1**: Success (Result::SUCCESS)
- * **0**: Failure (Result::FAILURE)
- * **-1**: Identify not found (Result::FAILURE_IDENTITY_NOT_FOUND)
- * **-2**: Credentials invalid (Result::FAILURE_CREDENTIAL_INVALID)
- * **-3**: Any other kind of failure (Result::FAILURE_OTHER)
- * **-4**: Credentials not found (Result::FAILURE_CREDENTIALS_NOT_FOUND)
+ * `'SUCCESS'`, value of `Result::SUCCESS`, when successful.
+ * `'FAILURE'`, value of `Result::FAILURE`, on any general failure.
+ * `'FAILURE_IDENTITY_NOT_FOUND'`, value of `Result::FAILURE_IDENTITY_NOT_FOUND`, when identify could not be found.
+ * `'FAILURE_CREDENTIALS_INVALID'`, value of `Result::FAILURE_CREDENTIALS_INVALID`, when credentials are invalid.
+ * `'FAILURE_CREDENTIALS_NOT_IN_REQUEST'`, value of `Result::FAILURE_CREDENTIALS_NOT_IN_REQUEST`, when credentials are not found in request.
```

Comment:
Isn't this effectively the same as the identity not being found?

Reply:
`FAILURE_IDENTITY_NOT_FOUND` is returned when an identity with given credentials hasn't been found by any identifier. `FAILURE_CREDENTIALS_NOT_FOUND` is returned when for example a username hasn't been found in post data, which usually means skipping the autenthicator and trying the next one.

Reply:
Can't we remove the FAILURE_ prefix as well? It's obvious that there is a problem if the identity was not found, same for the others.

Reply:
@markstory right because I was asking me this question I changed the constant name after reading a bit into the code base.

=> IDENTITY_NOT_FOUND => user does not exist
=> CREDENTIALS_NOT_IN_REQUEST => user credentials are missing in the request

Reply:
Just a remark, I'ts named 'IDENTITY' and not 'USER' for a reason: Not every actor in a system is an user. Could be a service or system as well that asks for authentication.